brush-parser/src/ast.rs

```diff
@@ -73,6 +76,30 @@ impl Display for AndOrList {
     }
 }
 
+/// Iterator for [`AndOrList`].
+pub struct AndOrListIter {
+    list: VecDeque<AndOr>,
+}
+
+impl Iterator for AndOrListIter {
+    type Item = AndOr;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        self.list.pop_front()
+    }
+}
+
+impl IntoIterator for AndOrList {
+    type Item = AndOr;
+    type IntoIter = AndOrListIter;
+
+    fn into_iter(self) -> Self::IntoIter {
+        let mut list = vec![AndOr::And(self.first)];
```

Comment:
I think the first item in the list could be either `And` or `Or` (or `None` and use `Option` for that matter), but whichever path we take we should find a way to document to callers what to expect.